.github/workflows/version-preview.yaml

```diff
@@ -0,0 +1,118 @@
+name: 'Version Preview'
+
+on:
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  preview:
+    name: Preview Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: git fetch origin ${{ github.base_ref }}
+
+      - name: Analyze PR Commits
+        id: analyze
+        run: |
+          # Get commits from this PR
+          COMMITS=$(git log origin/${{ github.base_ref }}..HEAD --pretty=format:"%s")
+          echo "Commits in this PR:"
```

Comment:
if we are using semantic release to determine this, why are we adding our own script for version? 
Why not use semantic release action itself?